site/app/libraries/FileUtils.php

```diff
@@ -73,6 +73,44 @@ public static function getAllFiles(string $dir, array $skip_files=[], bool $flat
         return $return;
     }
 
+    public static function getPathWithQueryAndTip($path, $query, $tip) {
+        // https://stackoverflow.com/questions/17160696/php-glob-scan-in-subfolders-for-a-file
+        $iter = new \RecursiveDirectoryIterator($path);
+        foreach (new \RecursiveIteratorIterator($iter) as $file){
+            if (strpos($file->getFilename() , $tip) !== false) {
+                if (preg_match("/.*" . $query . ".*" . $tip . "$/i", $file->getPathname()) > 0) {
+                    return $file->getPathname();
+                }
+            }
+        }
+        return false;
+    }
+
+    public static function getDetails($path, $query) {
+        $config_path = FileUtils::getPathWithQueryAndTip($path, $query, 'config.json');
+        $readme_path = FileUtils::getPathWithQueryAndTip($path, $query, 'README.md');
+        if ($config_path) {
+            $contents = FileUtils::json_decode_commented(file_get_contents($config_path), true);
+            $parsed_contents = array(
+                'path' => $config_path,
+                'title' => $contents['testcases'][0]['title'] ?? 'Title not Specified',
+                'tags' => $contents['tags'] ?? [],
+                'readme' => $readme_path,
+            );
+            return json_encode($parsed_contents);
+        }
+        return false;
+    }
+
+    // Credit to Alexander Shostak at https://stackoverflow.com/questions/8148797/a-json-parser-for-php-that-supports-comments
+    public static function json_decode_commented($json, $assoc = false, $maxDepth = 512, $opts = 0) {
```

Comment:
What is the reason for adding this?

Reply:
Because standard JSON code doesn't account for comments.

Reply:
JSON comment handling could be added to the `readJsonFile` function already in `FileUtils.php`.